libtenzir/src/execution_node.cpp

```diff
@@ -652,23 +652,21 @@ struct exec_node_state : inbound_state_mixin<Input>,
     if (paused or run_scheduled or not instance) {
       return;
     }
-    // We *always* use the delayed variant here instead of scheduling
-    // immediately as that has two distinct advantages:
-    // - It allows for using a weak actor pointer on the click, i.e., it does
-    //   not prohibit shutdown.
-    // - It does not get run immediately, which would conflict with operators
-    //   using `ctrl.self().request(...).await(...)`.
-    auto action = [this] {
-      auto time_scheduled_guard
-        = make_timer_guard(metrics->values.time_scheduled);
-      run_scheduled = false;
-      run();
-    };
     run_scheduled = true;
-    self->clock().schedule(self->clock().now(),
-                           caf::make_action(std::move(action),
-                                            caf::action::state::waiting),
-                           caf::weak_actor_ptr{self->ctrl()});
+    // It is critical that we send an _ordinary_ message to the scheduler here
+    // for the internal run loop, as opposed to a _system_ message, e.g., by
+    // scheduling an action directly on the clock. Otherwise, an operator using
```

Comment:
Tbf, I think scheduling directly on the clock is something that we will *never* want to do after the CAF updates, so it's probably not necessary to call this out in this particular part of the code.

However, the previous list of advantages still seems to be valid, so maybe we can leave them in as explanation.

Reply:
I removed the comment in 3e61cda7d0; in hindsight I think it adds more confusion than it helps.